public/react-services/wz-user-permissions.ts

```diff
@@ -950,33 +952,88 @@ export class WzUserPermissions{
-          : userPartialResources ? userPartialResources.some(resource => !isAllow(userPermissions[actionName][resource]))
-          : wazuhPermissions[actionName].resources.find(resource => resource === RESOURCE_ANY_SHORT) && userPermissions[actionName][RESOURCE_ANY] ? !isAllow(userPermissions[actionName][RESOURCE_ANY])
-          : true
+      const actionResourceAll = actionResource
+        .split('&')
+        .map(function (str) {
+          return str
+            .split(':')
+            .map(function (str, index) {
+              return index === 2 ? '*' : str;
+            })
+            .join(':');
+        })
+        .join('&');
+
+      const userPartialResources: string[] | undefined = userPermissions[actionName]
+        ? Object.keys(userPermissions[actionName]).filter((resource) =>
+            resource.match('&')
+              ? ![actionResource, actionResourceAll].includes(resource) &&
+                (resource.match(`/${actionResource}/`) || resource.match(`/${actionResourceAll}/`))
+              : ![actionResource, actionResourceAll].includes(resource) &&
+                (resource.match(actionResource.replace('*', '\\*')) ||
+                  resource.match(actionResourceAll.replace('*', '\\*')))
+          )
+        : undefined;
+
+      if (!userPermissions[actionName]) {
+        return userPermissions.rbac_mode === RBAC_MODE_WHITE;
       }
+
+      const existInWazuhPermissions = (userResource) => {
+        return !!wazuhPermissions[actionName].resources.find(function (resource) {
+          return (
+            resource ===
+            userResource
+              .split('&')
+              .map((str) => {
+                return str
+                  .split(':')
+                  .map(function (str, index) {
+                    return index === 2 ? '*' : str;
+                  })
+                  .join(':');
+              })
+              .join('&')
+              .replaceAll(':*', '')
+          );
+        });
+      };
+
+      const notAllowInWazuhPermissions = (userResource) => {
+        if (userResource !== RESOURCE_ANY) {
+          return existInWazuhPermissions(userResource)
+            ? !isAllow(userPermissions[actionName][userResource])
+            : true;
+        } else {
+          !isAllow(userPermissions[actionName][userResource]);
+        }
+      };
+
+      return userPermissions[actionName][actionResource]
+        ? notAllowInWazuhPermissions(actionResource)
+        : Object.keys(userPermissions[actionName]).some((resource) => {
+            return resource.match(actionResourceAll.replace('*', '\\*')) !== null;
+          })
+        ? Object.keys(userPermissions[actionName]).some((resource) => {
+            if (resource.match(actionResourceAll.replace('*', '\\*'))) {
+              return notAllowInWazuhPermissions(resource);
+            }
+          })
+        : userPartialResources?.length
```

Comment:
The App's `tsconfig.json` doesn't transpile the optional channing. We should modify this line or adapt the `tsconfig.json`